### DISCUSSION TITLE
How to add a <caption> after a <table> tag

### DISCUSSION THREAD

**Discussion by paulaami** [ueberdosis/tiptap#4907] [General Questions]
Hi, I use TipTap in Vue2, I need to add a button to editor which adding editable <caption> tag to a specific table.

Button adding table:
`<v-btn @click="editor.chain().focus().insertTable({ rows: 3, cols: 3, withHeaderRow: true }).run()"></v-btn>`

Button supossed to add <caption>
```
<v-btn @click="addCaption()" icon title="Add caption"</v-btn>
    addCaption() {
      const editor = this.editor
      const captionContent = '<caption contenteditable="true">test123</caption>'
      const currentContent = editor.getHTML()
      const tableIndex = currentContent.indexOf('<tbody>')

      if (tableIndex !== -1) {
        const updatedContent =
          currentContent.slice(0, tableIndex) +
          captionContent +
          currentContent.slice(tableIndex)
        editor.commands.setContent(updatedContent)
      } else {
        console.log('No <table> tag found in the editor content.')
      }
    },
```
Above code isnt working, How can I inject <caption> to look in DOM like `<table><caption>Place to add caption</caption><tbody></tbody></table>`

**Comment by hjerichen** [comment]
Did someone achieve to add a caption tag to the table?
